internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1142,6 +1143,13 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"snapshot_id": {
```

Comment:
should this be
```suggestion
			"create_from_snapshot_id": {
```

Reply:
I think we should align with the similar feature in node pool: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_node_pool_resource.go#L298